File: examples/csharp/mega_flow/Executor.cs
using System.Diagnostics;

namespace Mega;

public static class Executor
{
    // SINK — Process.Start · csharp.cmdi.process_start · CWE-78
    public static string Execute(string cmd)
    {
        Process.Start("cmd.exe", "/c " + cmd);
        return cmd;
    }

    public static string CleanTwin()
    {
        // NEGATIVE — same sink kind with a constant argument must not report.
        Process.Start("cmd.exe", "/c echo clean");
        return "clean";
    }
}

File: examples/typescript/mega_flow/executor.ts
import * as child_process from "child_process";

export function execute(cmd: string): unknown {
  // SINK — child_process.exec · CWE-78
  return child_process.exec(cmd);
}

export function clean_twin(): unknown {
  // NEGATIVE — same sink kind with a constant argument must not report.
  return child_process.exec("echo clean");
}
